packages/plugin-js-packages/src/lib/package-managers/derive-yarn.ts

```diff
@@ -0,0 +1,16 @@
+import { executeProcess } from '@code-pushup/utils';
+
+export async function deriveYarnVersion() {
+  const { stdout } = await executeProcess({
+    command: 'yarn',
+    args: ['-v'],
+  });
+
+  const yarnVersion = stdout.toString().trim().at(0);
+  if (yarnVersion === '2' || yarnVersion === '3') {
+    return 'yarn-modern';
+  } else if (yarnVersion === '1') {
+    return 'yarn-classic';
+  }
```

Comment:
The latest Yarn is version 4, which this code doesn't consider. For future-proofing, I would simply consider any version larger than 1 as Yarn Modern.

```suggestion
  const yarnVersion = Number.parseInt(stdout.toString().trim().at(0), 10);
  if (yarnVersion >= 2) {
    return 'yarn-modern';
  } else if (yarnVersion === 1) {
    return 'yarn-classic';
  }
```

Reply:
Thanks!! Way better. I have no clue why I did not plan future in :D